JoinColony/colonyCDapp
# [Advanced payments] feat: add annotation support to all expenditure sagas

## Description

Add annotation and custom title support to all expenditure sagas.

## How to test

Spin up a fresh environment/create a new colony.
You can also go to "Wayne enterprises" and click `Create expenditure`
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/b6f2ee38-6993-40b0-925b-9c7c6f217d9e)
Check that the transaction is pending and the last step should be `Annotating transaction`
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/3f63651e-6f66-4447-a39d-8fe2ddf0c147)

It should pass, check it by going to GraphiQL and running 
```
query MyQuery {
  listAnnotations {
    items {
      actionId
      message
    }
  }
}
```
And you should see annotations there!
![image](https://github.com/JoinColony/colonyCDapp/assets/23449297/6b00df19-cff5-477c-8089-872e55ce2dc1)


## Diffs

**New stuff** ✨

* Support for sending annotations and custom titles in all expenditure sagas

Resolves #1043
